### PR TITLE
Adds a signal for superior mobs reacting to attacks. Makes superior mobs target to whoever attacks t hem if they dont have a target. Updates signals a little.

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
Changes a lot of shit. Like, a lot. Mobs now search for targets and do see checks based on distance and viewrange rather than JUSt distance. Meaning, if a mob can see you but you're out of their viewrange, they'll act like they can't see you, and will lose you. 

Mobs now target whoever attacks them if they have no target. If they are out of viewrange, they will guess where the possible source of the attack came from by drawing a line towards and past the target, which is more inaccurate the further the target is. Added a few flags, one of which allows mobs to instantly move towards whoever attacked them.

I am so definitely refactoring all of mobcode after this.

## Changelog
:cl:
balance: Mobs now target you if you attack them, and will inaccurately guess your position if you shoot them from outside of their viewrange. Scope cheese is effectively neutrered.
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
WHY IS LINTERS FAILING WAHH

**Comment by nikothedude** [comment]
Why is it ?[]. why

**Comment by nikothedude** [comment]
maybe our linters are just outdated

**Comment by nikothedude** [comment]
Ready, although there might be a few unforseen bugs due to the scope.

Also, ignore the CI failure, if the server if 514 ?[] works.

File: code/modules/mob/living/carbon/superior_animal/attack.dm [hunk 1]
```diff
@@ -164,7 +164,7 @@
 			var/offset_temp = right_before_firing()
 			A.launch(target, def_zone, firer_arg = src, angle_offset = offset_temp) //this is where we actually shoot the projectile
 			right_after_firing()
-			SEND_SIGNAL(src, COMSIG_SUPERIOR_FIRED_PROJECTILE, src, A)
+			SEND_SIGNAL(src, COMSIG_SUPERIOR_FIRED_PROJECTILE, A)
```
Comment (Trilbyspaceclone) [inline code/modules/mob/living/carbon/superior_animal/attack.dm:167]: These are real linter fails

Comment (nikothedude) [reply code/modules/mob/living/carbon/superior_animal/attack.dm:167]: I disagree, I just think linters are outdated. Its not expecting the [ after the ?, despite ?[] being a valid operator introduced in 514.

**Pull request merged**